src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -140,6 +140,13 @@ private static Collection<IdCredentials> fetchCredentials() {
             if (StringUtils.isEmpty(keyVaultURL) || StringUtils.isEmpty(credentialID)) {
                 return Collections.emptyList();
             }
+
+            // If keyVaultURL does not have a trailing slash, add one
+            if (!String.valueOf(keyVaultURL.charAt(keyVaultURL.length() -1)).equals("/")) {
```

Comment:
I can't see a reason for the valueOf, is it necessary?
```suggestion
            if (!keyVaultURL.charAt(keyVaultURL.length() -1).equals("/")) {
```